### PR TITLE
[ENH] Sieve datetime operations

### LINKED ISSUE #1680: Sieve expert operations on DateTime fields
Feature request: Allow changing DateTime fields with relative values:

```
add! time.source = time.source - '1 hour'
```
To parse the last string, `intelmq.lib.utils.parse_relative` can be used.

### PR DESCRIPTION
**Pull request by waldbauer-certat**
Added possibility to use super basic math ( + and - operations ).

Sieve is now able to calculate a new datetime.

There should be no breaking change!

Usage:
```
add! time.observation += '1 hour'
add! time.observation -= '2 hours'

add time.observation += '1 hour'
add time.observation -= '2 hours'

update time.observation += '1 hour'
update time.observation -= '1 hour'
```

Closes #1680

### CODE REVIEW

File: docs/user/bots.rst [hunk 1]
```diff
@@ -2559,15 +2559,22 @@ specified with quotes. Following operators may be used to match events:
 If part of a rule matches the given conditions, the actions enclosed in `{` and
 `}` are applied. By default, all events that are matched or not matched by rules
 in the sieve file will be forwarded to the next bot in the pipeline, unless the
-`drop` action is applied.
+`drop` action is applied.
```
Comment (wagner-certat) [inline docs/user/bots.rst:2562]: space at line end

File: docs/user/bots.rst [hunk 2]
```diff
@@ -2559,15 +2559,22 @@ specified with quotes. Following operators may be used to match events:
 If part of a rule matches the given conditions, the actions enclosed in `{` and
 `}` are applied. By default, all events that are matched or not matched by rules
 in the sieve file will be forwarded to the next bot in the pipeline, unless the
-`drop` action is applied.
+`drop` action is applied. 
+
+**ATTENTION** Math only supported on DateTime objects currently!
```
Comment (wagner-certat) [inline docs/user/bots.rst:2564]: Please move the note to the section below where it belongs to (basic math operations)

File: docs/user/bots.rst [hunk 3]
```diff
@@ -2559,15 +2559,22 @@ specified with quotes. Following operators may be used to match events:
...
    ```add comment = 'hello, world'```
 
+   Basic math operations
+   ```add time.observation += '1 hour'```
+   ```add time.observation -= '10 hours'```
+
  * `add!` same as above, but will force overwrite the key in the event.
 
- * `update` modifies an existing value for a key. Only applies if the key is already defined. If the key is not defined in the event, this action is ignored.  Example:
+ * `update` modifies an existing value for a key. Only applies if the key is already defined. If the key is not defined in the event, this action is ignored. Supports mathematical expressions like above. Example:
```
Comment (wagner-certat) [inline docs/user/bots.rst:2577]: Can you add an example here as well please?

File: intelmq/bots/experts/sieve/expert.py [hunk 4]
```diff
@@ -230,6 +233,14 @@ def process_ip_range_match(self, key, ip_range, event):
                 if addr in network:
                     return True
         return False
+    
+    @staticmethod
+    def compute_basic_math(action, event):
+        date = dateutil.parser.parse(event[action.key])
```
Comment (wagner-certat) [inline intelmq/bots/experts/sieve/expert.py:239]: The event should have isoformat only, so no special parsing should be needed here.
See `DateTime.parse_utc_isoformat`

**Review by wagner-certat — CHANGES_REQUESTED** [review]
Thanks!
I have only some minor remarks

File: docs/user/bots.rst (hunk 2 above)
Comment (schacht-certat) [reply docs/user/bots.rst:2564]: The sentence is missing a verb.

File: docs/user/bots.rst (hunk 3 above)
Comment (schacht-certat) [reply docs/user/bots.rst:2577]: *This* only applies...
*This* supports...

**Review by schacht-certat — CHANGES_REQUESTED** [review]

**Review by wagner-certat — APPROVED** [review]

**Comment by wagner-certat** [comment]
I fixed the codestyle issues in 0a664b5cb

**Pull request merged**